redfish-core/lib/license_service.hpp

```diff
@@ -0,0 +1,678 @@
+                                   licenseString))
+    {
+        return;
+    }
+
+    if (licenseString.empty())
+    {
+        messages::propertyMissing(asyncResp->res, "LicenseString");
+    }
+
+    // Only allow one License install at a time
+    if (getLicenseActivationStatusMatch() != nullptr)
+    {
+        messages::resourceInUse(asyncResp->res);
+        return;
+    }
+
+    std::shared_ptr<boost::asio::steady_timer> timeout =
+        std::make_shared<boost::asio::steady_timer>(
+            crow::connections::systemBus->get_io_context());
+    timeout->expires_after(std::chrono::seconds(20));
+    crow::connections::systemBus->async_method_call(
+        [timeout, asyncResp,
+         licenseString](const boost::system::error_code& ec) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("LicenseString resp_handler got error {}", ec);
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        auto timeoutHandler =
+            [asyncResp, timeout](const boost::system::error_code& errCode) {
+            resetLicenseString(asyncResp);
+            if (errCode)
+            {
+                if (errCode != boost::asio::error::operation_aborted)
+                {
+                    BMCWEB_LOG_ERROR("Async_wait failed {}", errCode);
+                    messages::internalError(asyncResp->res);
+                    return;
+                }
+            }
+            else
+            {
+                BMCWEB_LOG_WARNING("Timed out waiting for HostInterface to "
+                                   "serve license upload request");
+                messages::serviceTemporarilyUnavailable(asyncResp->res, "60");
+                return;
+            }
+        };
+
+        timeout->async_wait(timeoutHandler);
+
+        auto callback = [asyncResp, timeout,
+                         licenseString](sdbusplus::message::message& m) {
+            BMCWEB_LOG_DEBUG("Response Matched ");
+            boost::container::flat_map<std::string, std::variant<std::string>>
+                values;
+            std::string iface;
+            m.read(iface, values);
+            if (iface == "com.ibm.License.LicenseManager")
+            {
+                auto findStatus = values.find("LicenseActivationStatus");
+                if (findStatus != values.end())
+                {
+                    BMCWEB_LOG_INFO("Found Status property change");
+                    std::string* status =
+                        std::get_if<std::string>(&(findStatus->second));
+                    if (status == nullptr)
+                    {
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+                    // Ignore D-bus propertyChanged signal for
+                    // status value "Pending"
+                    if (*status == "com.ibm.License.LicenseManager."
+                                   "Status.Pending")
```

Comment:
The new clang-format expects not to break a string
```
     if (*status == "com.ibm.License.LicenseManager.Status.Pending")
```

Reply:
did not get any format error

Reply:
done